crates/zizmor/src/audit/use_trusted_publishing.rs

```diff
@@ -201,6 +201,77 @@ audit_meta!(
 );
 
 impl UseTrustedPublishing {
+    // Detect whether this job is using crates.io Trusted Publishing by checking for
+    // a prior step that uses rust-lang/crates-io-auth-action and that this step
+    // wires its token into cargo via env or inline token usage.
+    fn uses_crates_io_trusted_publishing(
+        &self,
+        step: &crate::models::workflow::Step,
+        run: &str,
+    ) -> bool {
+        // Collect IDs of prior steps in the same job that use rust-lang/crates-io-auth-action
+        let mut auth_step_ids: Vec<&str> = Vec::new();
+
+        for prior in step.job().steps() {
+            if prior.index >= step.index {
+                break;
+            }
+
+            // Only consider uses: steps that reference the crates-io auth action
+            let Some(common_uses) = prior.uses() else { continue };
+            let github_actions_models::common::Uses::Repository(repo_uses) = common_uses else {
+                continue;
+            };
+
+            if repo_uses.owner.eq_ignore_ascii_case("rust-lang")
+                && repo_uses.repo.eq_ignore_ascii_case("crates-io-auth-action")
```

Comment:
Could you write this as a `RepositoryUsesPattern`? That'll ensure we fully match all of the different ways a user could write this `uses:` clause.

(It implements `FromStr`, so you can do `"rust-lang/crates-io-auth-action".parse().unwrap()` to get one.)